lib/internal/event_target.js

```diff
@@ -557,22 +558,30 @@ function emitUnhandledRejectionOrErr(that, err, event) {
   process.emit('error', err, event);
 }
 
+// A map of emitter -> map of name -> handler
+const eventHandlerValueMap = new SafeWeakMap();
```

Comment:
Is there a reason not to use a symbol property on the target object?

Reply:
Well, I didn't want to make any assumptions on the object I am receiving and a WeakMap is how to do that. A symbol property is probably fine since the object needs to not be frozen for defineProperty to work anyway. "security" of editing it is not a concern here anyway (since lots of other things are symbols and we don't make that guarantee).

So basically I'm ±0 on symbol vs WeakMap here and if you prefer I will happily do this with a symbol instead.

Reply:
Basically this change:
```diff
diff --git a/lib/internal/event_target.js b/lib/internal/event_target.js
index 3a7f156005..e78b958bca 100644
--- a/lib/internal/event_target.js
+++ b/lib/internal/event_target.js
@@ -14,7 +14,6 @@ const {
   Symbol,
   SymbolFor,
   SymbolToStringTag,
-  SafeWeakMap,
   SafeWeakSet,
 } = primordials;
 
@@ -36,6 +35,7 @@ const kIsEventTarget = SymbolFor('nodejs.event_target');
 const kEvents = Symbol('kEvents');
 const kStop = Symbol('kStop');
 const kTarget = Symbol('kTarget');
+const kHandlers = Symbol('khandlers');
 
 const kHybridDispatch = Symbol.for('nodejs.internal.kHybridDispatch');
 const kCreateEvent = Symbol('kCreateEvent');
@@ -558,27 +558,24 @@ function emitUnhandledRejectionOrErr(that, err, event) {
   process.emit('error', err, event);
 }
 
-// A map of emitter -> map of name -> handler
-const eventHandlerValueMap = new SafeWeakMap();
-
 function defineEventHandler(emitter, name) {
   // 8.1.5.1 Event handlers - basically `on[eventName]` attributes
   ObjectDefineProperty(emitter, `on${name}`, {
     get() {
-      return eventHandlerValueMap.get(this)?.get(name);
+      return this[kHandlers]?.get(name);
     },
     set(value) {
-      const oldValue = eventHandlerValueMap.get(this)?.get(name);
+      const oldValue = this[kHandlers]?.get(name);
       if (oldValue) {
         this.removeEventListener(name, oldValue);
       }
       if (typeof value === 'function') {
         this.addEventListener(name, value);
       }
-      if (!eventHandlerValueMap.has(this)) {
-        eventHandlerValueMap.set(this, new Map());
+      if (!this[kHandlers]) {
+        this[kHandlers] = new Map();
       }
-      eventHandlerValueMap.get(this).set(name, value);
+      this[kHandlers].set(name, value);
     },
     configurable: true,
     enumerable: true
```

Reply:
Yeah, I think that would be a bit cleaner :) I also think it would be safe to assume that the target is an `instanceof EventTarget` (i.e. subclass prototype or actual instance) anyway

Reply:
Done